synapse/http/client.py

```diff
@@ -819,6 +820,125 @@ def __init__(
             )
 
 
+class SimpleReplicationClient(BaseHttpClient):
+    """No frills client for connecting to Replication endpoints.
+
+    Uses existing BaseHttpClient methods but replaces the 'agent' used to make the
+    request with one that supports HTTP and HTTPS.
+    Attributes:
```

Comment:
Better yet, how about this?
```suggestion
    """No frills client for connecting to Replication endpoints via HTTP and HTTPS.
    
    Attributes:
```

Reply:
Errr... I edited the above instead of making a new suggestion. :( Sorry about that.